File: http/http/Helper/Auth/MyAuthService.cs
using http.Data;
using http.Data.Models;
using Microsoft.EntityFrameworkCore;
using System.Text.Json.Serialization;

namespace http.Helper.Auth
{
    public class MyAuthService
    {
        private readonly ApplicationDbContext _applicationDbContext;
        private readonly IHttpContextAccessor _httpContextAccessor;

        public MyAuthService(ApplicationDbContext applicationDbContext, IHttpContextAccessor httpContextAccessor)
        {
            _applicationDbContext = applicationDbContext;
            _httpContextAccessor = httpContextAccessor;
        }

        public bool IsLogiran()
        {
            return GetAuthInfo().isLogiran;
        }

        public bool IsAdmin()
        {
            return GetAuthInfo().korisnickiNalog?.isAdmin ?? false;
        }
        public bool IsKupac()
        {
            return GetAuthInfo().korisnickiNalog?.isKupac ?? false;
        }
        public bool IsVlasnik()
        {
            return GetAuthInfo().korisnickiNalog?.isVlasnik ?? false;
        }
        public bool IsRadnik()
        {
            return GetAuthInfo().korisnickiNalog?.isRadnik ?? false;
        }

        public MyAuthInfo GetAuthInfo()
        {
            string? authToken = _httpContextAccessor.HttpContext!.Request.Headers["my-auth-token"];

            AutentifikacijaToken? autentifikacijaToken = _applicationDbContext.AutentifikacijaToken
                .Include(x => x.korisnickiNalog)
                .SingleOrDefault(x => x.Vrijednost == authToken);

            return new MyAuthInfo(autentifikacijaToken);
        }

    }

    public class MyAuthInfo
    {
        public MyAuthInfo(AutentifikacijaToken? autentifikacijaToken)
        {
            this.autentifikacijaToken = autentifikacijaToken;
        }

        [JsonIgnore]
        public KorisnickiNalog? korisnickiNalog => autentifikacijaToken?.korisnickiNalog;
        public AutentifikacijaToken? autentifikacijaToken { get; set; }

        public bool isLogiran => korisnickiNalog != null;
    }
}

File: frontend/src/app/services/MyAuthService.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {AutentifikacijaToken} from "../helper/autentifikacijaToken";

@Injectable({providedIn:'root'})

export class MyAuthService{
  constructor() {
  }

  isLogiran():boolean{
    return this.getAuthorizationToken()!=null;
  }
  getAuthorizationToken():AutentifikacijaToken | null {
    let tokenString = window.localStorage.getItem("my-auth-token")??"";
    try {
      return JSON.parse(tokenString);
    }
    catch (e){
      return null;
    }
  }
  isAdmin():boolean {
    return this.getAuthorizationToken()?.korisnickiNalog.isAdmin ?? false
  }
  isKupac():boolean {
    return this.getAuthorizationToken()?.korisnickiNalog.isKupac ?? false
  }
  isRadnik():boolean {
    return this.getAuthorizationToken()?.korisnickiNalog.isRadnik ?? false
  }
  isVlasnik():boolean {
    return this.getAuthorizationToken()?.korisnickiNalog.isVlasnik ?? false
  }
  setLogiraniKorisnik(x: AutentifikacijaToken | null) {

    if (x == null){
      window.localStorage.setItem("my-auth-token", '');
    }
    else {
      window.localStorage.setItem("my-auth-token", JSON.stringify(x));
    }
  }
}
